lib/fog/azurerm/models/network/network_interface.rb

```diff
@@ -61,13 +68,57 @@ def save
           requires :subnet_id
           requires :ip_configuration_name
           requires :private_ip_allocation_method
-          network_interface = service.create_network_interface(resource_group, name, location, subnet_id, public_ip_address_id, ip_configuration_name, private_ip_allocation_method)
+          network_interface = service.create_or_update_network_interface(resource_group, name, location, subnet_id, public_ip_address_id, ip_configuration_name, private_ip_allocation_method, private_ip_address)
+          merge_attributes(Fog::Network::AzureRM::NetworkInterface.parse(network_interface))
+        end
+
+        def update(updated_attributes = {})
+          validate!(updated_attributes)
+          merge_attributes(updated_attributes)
+          network_interface = service.create_or_update_network_interface(resource_group, name, location, subnet_id, public_ip_address_id, ip_configuration_name, private_ip_allocation_method, private_ip_address)
```

Comment:
variable name should be more explicit, name it either
- network_interface_card or
- nic